build/yamls/antrea.yml

```diff
@@ -280,27 +295,49 @@ data:
             "type": "host-local"
         }
     }
-  antrea-controller.conf: ""
+  antrea-controller.conf: |
+    # Enable metrics exposure via Prometheus. Initializes Prometheus metrics listener
+    #enablePrometheusMetrics: false
+
+    # Enable golang metrics exposure via Prometheus
+    #enablePrometheusGoMetrics: false
+
+    # Enable process metrics exposure via Prometheus.
+    #enablePrometheusProcessMetrics: false
+
+    # Prometheus metrics hostname.
+    #prometheusHost:
+
+    # Prometheus listener port.
+    #prometheusPort: 9093
 kind: ConfigMap
 metadata:
   annotations: {}
   labels:
     app: antrea
-  name: antrea-config-4gb24b784b
+  name: antrea-config-dbb9d5fhf9
   namespace: kube-system
 ---
 apiVersion: v1
 kind: Service
 metadata:
+  annotations:
+    prometheus.io/port: "9093"
+    prometheus.io/scrape: "true"
   labels:
     app: antrea
   name: antrea
   namespace: kube-system
 spec:
   ports:
-  - port: 443
+  - name: https
+    port: 443
     protocol: TCP
     targetPort: 443
+  - name: monitoring
+    port: 9093
+    protocol: TCP
+    targetPort: 9093
```

Comment:
I found that there's already a metrics api exposed by starting apiserver. Should we just reuse that one and just add antrea's own metrics items?
See the output by accessing antrea-agent's api (I think antrea-controller has it too but need to skip authentication for this path:
```
curl -k https://127.0.0.1:10443/metrics
# TYPE apiserver_audit_event_total counter
apiserver_audit_event_total 0
# HELP apiserver_audit_requests_rejected_total Counter of apiserver requests rejected due to an error in audit logging backend.
# TYPE apiserver_audit_requests_rejected_total counter
apiserver_audit_requests_rejected_total 0
# HELP apiserver_client_certificate_expiration_seconds Distribution of the remaining lifetime on the certificate used to authenticate a request.
# TYPE apiserver_client_certificate_expiration_seconds histogram
apiserver_client_certificate_expiration_seconds_bucket{le="0"} 0
apiserver_client_certificate_expiration_seconds_bucket{le="1800"} 0
```

Reply:
Makes sense. Will give that a try.

Reply:
@tnqn this seem to work for the agent but not for the controller. Any suggestion or idea?

Reply:
It's restricted by default authorization strategy, make the controller apiserver skip authorization for "/metrics" can solve it.
https://github.com/vmware-tanzu/antrea/blob/master/cmd/antrea-controller/controller.go#L111
```
	authorization := genericoptions.NewDelegatingAuthorizationOptions()
	authorization.WithAlwaysAllowPaths("/metrics")
```

Reply:
Indeed it works...